app/lib/clients/vault.rb

```diff
@@ -0,0 +1,48 @@
+module Clients
+  class Vault
+    class << self
+      def issue_cert(cert_issue_request)
+        opts = cert_issue_request.attributes
+        # Generate the TLS certificate using the intermediate CA
+        tls_cert = client.logical.write(Rails.configuration.astral[:vault_cert_path], opts)
+        OpenStruct.new tls_cert.data
+      end
+
+      def kv_read(path)
+        client.kv(kv_mount).read(path)
+      end
+
+      def kv_write(path, data)
+        enable_engine(kv_mount, "kv-v2")
+        client.logical.write("#{kv_mount}/data/#{path}", data: data)
+      end
+
+      def kv_delete(path)
+        client.logical.delete("#{kv_mount}/data/#{path}")
+      end
+
+      private
+
+      def client
+        # TODO create a new token for the session
```

Comment:
Are there tickets for the TODOs in this PR?

Reply:
i think that's been around for a while but I moved the file. I'll delete TODO and make tix

Reply:
https://github.com/G-Research/astral/issues/34